src/service/rankingService.js

```diff
@@ -0,0 +1,99 @@
+let user =  {
+  name: 'Player1',
+  score: 15,
+  timeInSeconds: 93
+};
+// check localStorage for current Top scores
+// if empty create object with objects mode1, mode2, and mode3
+// PokemonApiRanking - data from/to localStorage
+const PokemonApiRanking = JSON.parse(localStorage.getItem('PokemonApiRanking')) || {
+  mode1: {
+    name : "Whos that pokemon?",
+    scores : []
+  },
+  mode2: {
+    name : "What it looks like?",
+    scores : []
+  },
+  mode3: {
+    name : "Guess the type!",
+    scores : []
+  }
+};
+
+export const rankingService = (mode, score) => {
+    
+  let currentMode; //path to scores array for current mode
+  switch(mode) {
+    case WHO_IS_THAT_POKEMON:
+      currentMode = PokemonApiRanking.mode1.scores;
+    break;
+    case WHAT_DOES_THIS_POKEMON_LOOK_LIKE:
+      currentMode = PokemonApiRanking.mode2.scores;
+    break;
+    case GUESS_THE_TYPE:
+      currentMode = PokemonApiRanking.mode3.scores;
+    break;
+    default:
+      trow `Wrong mode`;
+  }
+  currentMode.push(user);
+  currentMode.sort( (a,b) => b.timeInSeconds - a.timeInSeconds );
+  currentMode.sort( (a,b) => b.score - a.score );
+  if(currentMode.length > 3) {
+    currentMode.length = 3; 
+  }
```

Comment:
this approach looks better for me

Reply:
but setting array length is not good 🤮

Reply:
best thing would be to:
1. add new record to old array
2. sort it
3. create new array
4. add only 3 first elements of old array to new array
5. set "new array" as right property of whole PokemonApiRanking